### PR TITLE
Improve tests for batched transactions

### PR DESCRIPTION
**Pull request by sideninja**
## Description
Improve testing of batched transactions.
______

For contributor use:

- [ ] Targeted PR against `master` branch
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [standards mentioned here](https://github.com/onflow/flow-nft/blob/master/CONTRIBUTING.md#styleguides).
- [ ] Updated relevant documentation 
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Added appropriate labels 

## Summary by CodeRabbit

- **New Features**
  - Enhanced the batch processing of Ethereum value transfer transactions in end-to-end tests.

- **Tests**
  - Improved test logic to handle multiple value transfers and batch transactions.
  - Updated test functions to create and manage batch transactions more efficiently.
  - Refined batch retrieval of transaction details and assertions.

These changes improve the robustness and efficiency of our testing processes, ensuring better performance and reliability.

### CODE REVIEW

File: tests/web3js/eth_batch_retrieval_test.js [hunk 1]
```diff
@@ -4,19 +4,22 @@ const helpers = require('./helpers')
...
+    let batchSize = 10
+    assert.lengthOf(block.transactions, batchSize)
 
-    let callTx = await web3.eth.getTransactionFromBlock(latestHeight, 1)
-    assert.equal(block.number, callTx.blockNumber)
-    assert.equal(block.hash, callTx.blockHash)
-    assert.equal(0, callTx.type)
-    assert.equal(1, callTx.transactionIndex)
+    for (let i = 0; i < block.transactions.length; i++) {
+        let tx = await web3.eth.getTransactionFromBlock(latestHeight, block.number)
```
Comment (m-Peter) [inline tests/web3js/eth_batch_retrieval_test.js:16]:
```suggestion
        let tx = await web3.eth.getTransactionFromBlock(latestHeight, i)
```

File: tests/web3js/eth_batch_retrieval_test.js [hunk 2]
```diff
@@ -4,19 +4,22 @@ const helpers = require('./helpers')
...
+        console.log(tx)
+        console.log("-----")
+        assert.equal(tx.blockNumber, block.number, "wrong block number")
+        assert.equal(tx.blockHash, block.hash, "wrong block hash")
+        assert.equal(tx.type, 0, "wrong type")
+        //assert.equal(tx.transactionIndex, i, "wrong index")
```
Comment (m-Peter) [inline tests/web3js/eth_batch_retrieval_test.js:22]:
```suggestion
        assert.equal(tx.transactionIndex, i, "wrong index")
```

File: tests/web3js/eth_batch_retrieval_test.js [hunk 3]
```diff
@@ -4,19 +4,22 @@ const helpers = require('./helpers')
...
 
-    let callTx = await web3.eth.getTransactionFromBlock(latestHeight, 1)
-    assert.equal(block.number, callTx.blockNumber)
-    assert.equal(block.hash, callTx.blockHash)
-    assert.equal(0, callTx.type)
-    assert.equal(1, callTx.transactionIndex)
+    for (let i = 0; i < block.transactions.length; i++) {
+        let tx = await web3.eth.getTransactionFromBlock(latestHeight, block.number)
+        console.log(tx)
+        console.log("-----")
```
Comment (m-Peter) [inline tests/web3js/eth_batch_retrieval_test.js:18]: I believe we can remove these 2 `console.log` statements.

**Review by m-Peter — APPROVED** [review]
LGTM!

File: tests/e2e_web3js_test.go [hunk 4]
```diff
@@ -37,40 +42,48 @@ func TestWeb3_E2E(t *testing.T) {
 	})
 
 	t.Run("batch run transactions", func(t *testing.T) {
-		runWeb3TestWithSetup(t, "eth_batch_retrieval_test", func(emu emulator.Emulator) error {
-			tx1, err := cadence.NewString("f9015880808301e8488080b901086060604052341561000f57600080fd5b60eb8061001d6000396000f300606060405260043610603f576000357c0100000000000000000000000000000000000000000000000000000000900463ffffffff168063c6888fa1146044575b600080fd5b3415604e57600080fd5b606260048080359060200190919050506078565b6040518082815260200191505060405180910390f35b60007f24abdb5865df5079dcc5ac590ff6f01d5c16edbc5fab4e195d9febd1114503da600783026040518082815260200191505060405180910390a16007820290509190505600a165627a7a7230582040383f19d9f65246752244189b02f56e8d0980ed44e7a56c0b200458caad20bb002982052fa09c05a7389284dc02b356ec7dee8a023c5efd3a9d844fa3c481882684b0640866a057e96d0a71a857ed509bb2b7333e78b2408574b8cc7f51238f25c58812662653")
```
Comment (m-Peter) [inline tests/e2e_web3js_test.go:41]: I know that it's impossible to understand what these 2 transactions did :sweat_smile: 
I believe it was a good test case scenario though, it was basically a contract deployment, followed by a contract function call. But I suppose that such logic must already have been tested against in the `flow-go` repository.

Comment (sideninja) [reply tests/e2e_web3js_test.go:41]: I plan to add more complex transactions as well, but I prefer actually building them not just passing payload, that way is clear what they do and it's  easier to change.

**Review by m-Peter — APPROVED** [review]

**Pull request merged**